lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1520,12 +1528,18 @@ ChangeResult TypeAnalyzer::visitAtenTensorOp(AtenTensorOp op) {
 }
 
 template <typename OpTy>
-ChangeResult TypeAnalyzer::visitConstantTensorAllocOp(OpTy op) {
+ChangeResult
+TypeAnalyzer::visitConstantTensorAllocOp(OpTy op,
+                                         llvm::Optional<Type> inputDType) {
   auto knowledge =
       ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
   fillInSizesGivenSizesList(knowledge, op.size());
-  fillInDTypeGivenDTypeAndDataType(knowledge, op.dtype(),
-                                   Torch::FloatType::get(op->getContext()));
+  if (isa<AtenFullOp>(op))
+    fillInDTypeGivenDTypeIntAndInputDType(knowledge, op.dtype(),
```

Comment:
not related to this PR but can you help to add an assertion for `fillInDTypeGivenDTypeIntAndInputDType ` to assert the `inputDType` is from builtinDialect? Similar to this assert 
https://github.com/llvm/torch-mlir/blob/ba29d4f25041848ac96d4cd54aa9a23d81d77f1a/lib/Dialect/Torch/Transforms/RefineTypes.cpp#L757-L758